graviti/schema/field.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+"""Parameter typy releated classes."""
+
+
+from typing import Any, Dict, Iterable, List, Mapping, NamedTuple, Sequence, Tuple, Union, overload
+
+from graviti.schema.base import _INDENT, PortexType
+
+
+class Field(NamedTuple):
+    """Represents a Portex ``record`` field, contains the field name and type."""
+
+    name: str
+    type: PortexType
+
+
+class Fields(Sequence[Field]):
+    """Represents a Portex ``record`` field list."""
+
+    def __init__(
+        self,
+        fields: Union[Iterable[Union[Field, Tuple[str, PortexType]]], Mapping[str, PortexType]],
+    ):
+        iterable = fields.items() if isinstance(fields, Mapping) else fields
+        self._data = [Field(*field) for field in iterable]  # pyright: reportGeneralTypeIssues=false
```

Comment:
Do we need to block pyright errors?

Reply:
This is quite complicate, we support two input types:
A: `Iterable[Tuple[str, PortexType]]` 
B: `Mapping[str, PortexType]`

The problem is `Mapping is Iterable`, which means if user input a  Mapping which key is `Tuple[str, PortexType]`, it will pass the type check, but can not be property handled by code. That is why pyright report error.

Python typing system can not specify a type which is `Iterable` but not `Mapping`.
I can not find a perfect solution so far.